### ISSUE TITLE
201801196_Challenges_Solutions 

### ISSUE THREAD

**Issue by khyatibhuva** [nisargptl/vehicle-recognition#63]
Write your challenges you faced in this entire project and how to solved them.

**Comment by 201801196** [comment]
We had a dying relu problem in VGG model. we solved the problem by using feature visualization. We extracted the features from last layers and found the layers which did not have activated during gradient ascent. due to which is resulted in noise. so we concluded that many neurons in our model died because of the properties of relu it can not learn new things any more. the possible solution were reassigning the weights which would lead to loss in pretrained VGG model. So we decided to train the model again with penalty and smaller learning rate and smaller batch sizes. 

we had a problem with the moblinet model, In which during training due to some technical problem our training stopped and our model got corrupted so we had an accuracy of 87% at that time and the model gave around 28% accuracy. so we had to train the model again.  This isssue was resolved by retraining the model. We started using epoch-wise checkpoint to make sure this type of problem doesn't occur again. 

we also had a problem while using the cutout augmentation. Standard machine learning practice is to use the image normalized between [0,1] but mobilenet uses [-1, 1]. we the cutout augmentation library used [0,1] type of normalization. so our model predicted invalid classes valid with high probability. we solved this problem by augmenting before we create the tensoflow dataflow pipeline.
